packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -773,136 +756,84 @@ export class IntervalCollectionView<TInterval extends ISerializableInterval> ext
-    }
-
-    public next() {
-        let _value: TInterval | undefined;
-        let _done: boolean = true;
-
-        if (this.index < this.results.length) {
-            _value = this.results[this.index++];
-            _done = false;
-        }
-
-        return {
-            value: _value,
-            done: _done,
-        };
-    }
-}
-
-export class IntervalCollection<TInterval extends ISerializableInterval> {
-    private savedSerializedIntervals?: ISerializedInterval[];
-    private view: IntervalCollectionView<TInterval>;
-
-    public get attached(): boolean {
-        return !!this.view;
-    }
-
-    constructor(private readonly helpers: IIntervalHelpers<TInterval>, private readonly requiresClient: boolean,
-        private readonly emitter: IValueOpEmitter,
-        serializedIntervals: ISerializedInterval[]) {
-        this.savedSerializedIntervals = serializedIntervals;
-    }
-
-    public attachGraph(client: MergeTree.Client, label: string) {
-        if (this.view) {
-            throw new Error("Only supports one Sequence attach");
-        }
-
-        if ((client === undefined) && (this.requiresClient)) {
-            throw new Error("Client required for this collection");
-        }
-
-        this.view = new IntervalCollectionView<TInterval>(client,
-            this.savedSerializedIntervals, label, this.helpers, this.emitter);
-        this.savedSerializedIntervals = undefined;
-    }
-
-    public add(
-        startPosition: number,
-        endPosition: number,
-        intervalType: MergeTree.IntervalType,
-        props?: MergeTree.PropertySet,
-    ) {
-        if (!this.view) {
-            throw new Error("attach must be called prior to adding intervals");
-        }
-
-        return this.view.add(startPosition, endPosition, intervalType, props);
-    }
-
-    public delete(
-        startPosition: number,
-        endPosition: number,
-    ) {
-        if (!this.view) {
-            throw new Error("attach must be called prior to deleting intervals");
-        }
-
-        this.view.delete(startPosition, endPosition);
-    }
 
-    public addConflictResolver(conflictResolver: MergeTree.IntervalConflictResolver<TInterval>): void {
-        this.view.addConflictResolver(conflictResolver);
-    }
-
-    public async getView(onDeserialize?: DeserializeCallback): Promise<IntervalCollectionView<TInterval>> {
-        if (!this.view) {
-            return Promise.reject(new Error("attachSequence must be called prior to retrieving the view"));
+    /**
+     * @deprecated - IntervalCollectionView has been removed. Please refer to IntervalCollection directly.
+     */
+    public getView(onDeserialize?: DeserializeCallback): IntervalCollection<TInterval> {
+        if (!this.attached) {
+            throw new Error("attachSequence must be called prior to retrieving the view");
         }
 
         // Attach custom deserializers if specified
         if (onDeserialize) {
```

Comment:
>if [](http://example.com/codeflow?start=8&length=2)

what is this? seem stange we need to do anything in this method

Reply:
It does. I don't see a case where a value is passed for this parameter.